src/librustdoc/html/render/print_item.rs

```diff
@@ -1560,16 +1571,37 @@ fn document_type_layout(w: &mut Buffer, cx: &Context<'_>, ty_def_id: DefId) {
                  <a href=\"https://doc.rust-lang.org/reference/type-layout.html\">“Type Layout”</a> \
                  chapter for details on type layout guarantees.</p></div>"
             );
-            if ty_layout.layout.abi.is_unsized() {
-                writeln!(w, "<p><strong>Size:</strong> (unsized)</p>");
-            } else {
-                let bytes = ty_layout.layout.size.bytes();
-                writeln!(
-                    w,
-                    "<p><strong>Size:</strong> {size} byte{pl}</p>",
-                    size = bytes,
-                    pl = if bytes == 1 { "" } else { "s" },
-                );
+            w.write_str("<p><strong>Size:</strong> ");
+            write_size_of_layout(w, ty_layout.layout, 0);
+            writeln!(w, "</p>");
+            if let Variants::Multiple { variants, tag, .. } = &ty_layout.layout.variants {
+                if !variants.is_empty() {
+                    w.write_str(
+                        "<p>\
+                            <strong>Size for each variant:</strong>\
+                            <ul>",
```

Comment:
I think `<ul>` is not allowed to be nested inside `<p>`. Can you change it to something like this?

```suggestion
                        "<p><strong>Size for each variant:</strong></p>\
                         <ul>",
```

and then remove the `</p>` lower down?